README.md

```diff
@@ -1,3 +1,262 @@
+
+TS-Minify *only*  targets property renaming. Minification tactics like whitespace removal, dead code removal, variable name mangling, etc. are taken care of by tools such as [UglifyJS](https://github.com/mishoo/UglifyJS). TS-Minify specifically targets property renaming since it is something that is difficult to achieve *safely* without type information. As such, TS-Minify requires a program to be correctly and throughly typed, otherwise, unwanted renaming may occur. 
+
+###How Does it Work? The TL;DR
+
+---
+The TypeScript Compiler API is utilized in order to access the Abstract Syntax Tree of a TypeScript source file. Identifiers that might be considered properties (such as identifiers in property declarations, property access expressions, method names in method declarations, etc.) are renamed to short names such as `a`, `b`, etc. after determining their renaming eligibility. A “renaming” global map is created with mappings from the original property name to the new generated property name so that re-namings are kept consistent between properties with the same names.
+
+```javascript
+{ maybeHandleCall: g }
+{ reportMissingType: h }
+{ getHandler: i }
+{ handlePropertyAccess: j }
+{ emitExtraImports: k }
+{ emit: l }
+{ visitTypeName: m }
+```
+<sub>An example of some mappings from the original property name to the shorter, generated property name.</sub>
+
+The renaming eligibility of a property takes several factors into consideration:
+
+- Does the property belong to an object declared in an external file ([`.d.ts file`](http://definitelytyped.org/))?
+- Does the property belong to a [standard built-in object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects)?
+- Does the property name belong to the DOM?
+- Does the property belong to an object being exported?
+
+###External vs. Internal Types
+
+---
+TS-Minify is able to safely rename properties by understanding which types are *external*, and which are not, to one's program.
+
+An *external* type is something that you are bringing into your program. It is a type which you have not defined. This includes objects and functions from the JavaScript standard library, and JavaScript libraries like underscore.js or jQuery. If you are using external libraries, it is imperative that you include their corresponding `.d.ts` files.
+
+Typings for the standard library are included in the default library typings, `lib.d.ts`, which the TypeScript compiler uses during transpilation (it will typecheck your program at compile time) and does not need to be explicitly included.
+
+*Internal* types are ones that you, the programmer, have defined yourselves. Declaring a class `Foo` in your program tells the minifier that properties on `Foo` objects are within the renaming scope. Internal types also include things like object literal types.
```

Comment:
Add something that reminds people that they need to feed all internal sources into a single minification pass.